botocore/serialize.py

```diff
@@ -786,20 +846,55 @@ def _serialize_type_boolean(self, xmlnode, params, shape, name):
         else:
             str_value = 'false'
         node.text = str_value
+        self._add_xml_namespace(shape, node)
 
     def _serialize_type_blob(self, xmlnode, params, shape, name):
         node = ElementTree.SubElement(xmlnode, name)
         node.text = self._get_base64(params)
+        self._add_xml_namespace(shape, node)
 
     def _serialize_type_timestamp(self, xmlnode, params, shape, name):
         node = ElementTree.SubElement(xmlnode, name)
-        node.text = self._convert_timestamp_to_str(
-            params, shape.serialization.get('timestampFormat')
+        node.text = str(
```

Comment:
Why are we casting this to a string here?  Does `_convert_timestamp_to_str` not always return a string?

Reply:
Yea, this name is misleading. I've updated it to `_convert_timestamp` and will include in the next revision.